### ISSUE TITLE
[BUG] Pod network theoretical IPs calculations blocks cluster autoscaler

### ISSUE THREAD

**Issue by slawekww** [Azure/AKS#4467]
**Describe the bug**

AKS with Azure CNI overlay.
Based on docs: https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay
each node takes 254 IPs from pod network.

Lets take example of AKS cluster where `pod_cidr` is defined as 192.168.128.0/17 
as only address space is available: 192.168.0.0/16 (private space).
The 2nd block is required for `service_cidr` like 192.168.0.0/17.
Other private network ranges may be reserved for other non-AKS purpose like big address space 10.0.0.0/8.

In this case only 32K pod IPs are available.

Formula to calculate theoretical max IP usage is:
```
theoretical max = sum of ((max count per node pool+1) * 254)
```

Need to add +1 for node pool upgrade.
In this case, max nodes for AKS is **128** nodes only (from possible 5K!).

Problem is that it is not possible to add more node pools or increase max count for nodepool
once theoretical max IPs exceeds pod IPs availability. 

AKS is scalable system and theoretical max should not block AKS configuration updates - adding or update node pools.
This max should be displayed by advisor and scale up should be blocked once IPs are exhausted in **reality**, not theoretically. 

**To Reproduce**
1. Create AKS cluster with pod_cidr = `192.168.128.0/17` and service_cidr=`192.168.0.0/17`
2. add node pool with max node count = 200

**Expected behavior**
This max should be displayed by Advisor and scale up should be blocked once IPs are exhausted in **reality**, not theoretically. 
Theoretical max check is blocked to define scalable system where in max of all nodes may never happen together. 

**Screenshots**
Example error:
```
Agent Pool Name: "test"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with response: {
│   "code": "InsufficientSubnetSize",
│   "details": null,
│   "message": "Pre-allocated IPs 36864 exceeds IPs available 32768 in Subnet Cidr 192.168.128.0/17, Subnet Name networkProfile.podCIDR. If Autoscaler is enabled, the max-count from each nodepool is counted towards this total (which means that pre-allocated IPs count represents a theoretical max value, not the actual number of IPs requested). [http://aka.ms/aks/insufficientsubnetsize",](http://aka.ms/aks/insufficientsubnetsize%22,)
│   "subcode": "",
│   "target": "networkProfile.podCIDR"
```

**Environment (please complete the following information):**
 -  CLI Version: 2.62.0
 - Kubernetes version: 1.29.4
 - CLI Extension version: N/A
 - Browser: N/A

**Additional context**
It is not cost efficient system to configure max only based on pod network availability.

**Comment by slawekww** [comment]
As AKS user, I would like to **opt-in** to choose if node scale up would fail if no pod network IPs are available instead of blocking changing max count on existing node pools or adding new node pools based on theoretical formula.
Dynamic Podsubnet would not work for my case as Azure CNI Overlay does not support it as good I understand - it assigns always 254 IPs per node based on docs. 

Please note:
- formula is not documented on any AKS docs
- theoretical max value has been introduced this year (I do not know exactly Apr-May timeframe) and impacts existing clusters which stopped operation to scale up node pools due cluster autoscaler issue as mentioned in this issue

To sum up, it is also feedback how this feature has been introduced to AKS.

**Comment by slawekww** [comment]
Lets close it as I've found way to use bigger subnets for AKS clusters.
